controllers/authorino_controller.go

```diff
@@ -443,73 +478,96 @@ func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino
-				return r.wrapErrorWithStatusUpdate(
-					logger, authorino, r.setStatusFailed(api.AuthorinoUnableToCreateServiceAccount),
-					fmt.Errorf("failed to create %s ServiceAccount, err: %v", saName, err),
-				)
-			}
+	if sa, err := r.createAuthorinoServiceAccount(authorino); err != nil {
+		return err
+	} else {
+		// creates the ClusterRoleBinding/RoleBinding depending on type of installation
+		if err := r.reconcileAuthorinoRoleBinding(authorinoManagerClusterRoleBindingNameSuffix, authorinoClusterRoleName, *sa, authorino.Spec.ClusterWide, authorino); err != nil {
+			return err
 		}
-		return r.wrapErrorWithStatusUpdate(
-			logger, authorino, r.setStatusFailed(api.AuthorinoUnableToGetServiceAccount),
-			fmt.Errorf("failed to get %s ServiceAccount, err: %v", saName, err),
-		)
+		// creates the K8s Auth ClusterRoleBinding (for Authorino's Kubernetes TokenReview and SubjectAccessReview features)
+		// Disclaimer: this has nothing to do with kube-rbac-proxy, but to authn/authz features of Authorino that also require cluster scope role bindings
+		if err := r.reconcileAuthorinoRoleBinding(authorinoK8sAuthClusterRoleBindingNameSuffix, authorinoK8sAuthClusterRoleName, *sa, true, authorino); err != nil {
+			return err
+		}
+
+		// creates leader election role
+		// TODO: describe what this leader election is
+		return r.leaderElectionPermission(authorino, *sa)
 	}
+}
 
-	// creates the ClusterRoleBinding/RoleBinding depending on type of installation
-	// if ClusterWide true - ClusterRoleBinding
-	// if ClusterWide false - RoleBinding
-	var binding client.Object = authorinoResources.GetAuthorinoClusterRoleBinding(authorinoClusterRoleName, sa.Name, sa.Namespace)
-	if !authorino.Spec.ClusterWide {
-		binding = authorinoResources.GetAuthorinoRoleBinding(authorinoClusterRoleName, sa.Name, sa.Namespace)
-		binding.SetNamespace(authorino.Namespace)
+func authorinoRoleBindingName(crName, roleBindingNameSuffix string, clusterScoped bool) string {
+  if clusterScoped {
+		return roleBindingNameSuffix
+	} else {
+		return fmt.Sprintf("%s-%s", crName, roleBindingNameSuffix)
+	}
+}
+
+func (r *AuthorinoReconciler) reconcileAuthorinoRoleBinding(roleBindingNameSuffix, roleName string, serviceAccount k8score.ServiceAccount, clusterScoped bool, authorino *api.Authorino) error {
+	var logger = r.Log
+
+	var roleBinding client.Object
+	var roleBindingName = authorinoRoleBindingName(authorino.Name, roleBindingNameSuffix, clusterScoped)
+	var roleBindingSearchName types.NamespacedName
+
+	if clusterScoped {
+		roleBinding = authorinoResources.GetAuthorinoClusterRoleBinding(roleName, serviceAccount)
+		roleBindingSearchName = types.NamespacedName{Name: roleBindingName}
+	} else {
+		roleBinding = authorinoResources.GetAuthorinoRoleBinding(roleName, serviceAccount)
+		roleBinding.SetNamespace(authorino.Namespace)
+		roleBindingSearchName = types.NamespacedName{Name: roleBindingName, Namespace: authorino.Namespace}
```

Comment:
We could use the [namespacedName](https://github.com/Kuadrant/authorino-operator/blob/788fd2809c021eef15f1dac6b1a2703d95ff46d2/controllers/authorino_controller.go#L723) function to assigned the value to the `roleBindingSearchName` variable

Reply:
Done.